src/rpcq.lisp

```diff
@@ -66,8 +66,14 @@ The input strings are assumed to be FORMAT-compatible, so sequences like ~<newli
 (defmethod %serialize (payload)
   payload)
 
+(defmethod %serialize ((payload array))
+  (map 'vector #'%serialize payload))
+
 (defmethod %serialize ((payload cons))
-  (loop :for elt :in payload :collect (%serialize elt)))
+  (map 'vector #'%serialize payload))
```

Comment:
can we nix serializing conses altogether?

Reply:
If you were to ask me: yes. If you were to ask someone else with more principles about backwards compatibility: no, or not yet.